unstructured_ingest/v2/processes/connectors/github.py

```diff
@@ -0,0 +1,357 @@
+                    ),
+                    metadata=FileDataSourceMetadata(
+                        url=element.url,
+                        version=element.etag,
+                        record_locator=record_locator,
+                        date_modified=date_modified,
+                        date_created=date_created,
+                        date_processed=str(time()),
+                    ),
+                    additional_metadata=additional_metadata,
+                )
+
+
+class GitHubDownloaderConfig(DownloaderConfig):
+    pass
+
+
+@dataclass
+class GitHubDownloader(Downloader):
+    connection_config: GitHubConnectionConfig
+    download_config: GitHubDownloaderConfig
+
+    def is_async(self) -> bool:
+        return True
+
+    @requires_dependencies(["github"], extras="github")
+    def _fetch_file(self, path: str) -> ContentFile:
+        """Fetches a file from the GitHub repository using the GitHub API.
+
+        Args:
+            path (str): The path to the file in the repository.
+
+        Returns:
+            ContentFile: An object containing the file content.
+
+
+        Raises:
+            UnknownObjectException: If the file does not exist in the repository.
+        """
+        from github.GithubException import UnknownObjectException
+
+        try:
+            logger.info(f"Fetching file from path: {path!r}")
+            content_file = self.connection_config.get_repo().get_contents(path)
+        except UnknownObjectException:
+            logger.error(f"File doesn't exist: {self.connection_config.url}/{path}")
+            raise UnknownObjectException
+
+        return content_file
+
+    @SourceConnectionNetworkError.wrap
+    @requires_dependencies(["httpx", "github"], extras="github")
+    async def _fetch_content(self, content_file: ContentFile) -> bytes:
+        """Asynchronously retrieves the content of a file, handling large files via direct download.
+
+        Args:
+            content_file (ContentFile): The file object from the GitHub API.
+
+        Returns:
+            bytes: The content of the file as bytes.
+
+        Raises:
+            UnknownObjectException: If the download or content retrieval fails.
+        """
+        import httpx
+        from github.GithubException import UnknownObjectException
+
+        contents = b""
+        async with httpx.AsyncClient() as client:
+            if not content_file.content and content_file.encoding == "none" and content_file.size:
+                logger.info(
+                    "File too large for the GitHub API, using direct download link instead."
+                )
+                try:
+                    response = await client.get(content_file.download_url, timeout=10.0)
+                    response.raise_for_status()
+                except (httpx.HTTPStatusError, httpx.RequestError) as e:
+                    logger.error(f"Failed to download {content_file.download_url}: {e}")
+                    raise UnknownObjectException
```

Comment:
Can we catch any error, log it, and then reraise the same error to not lose the context:
```suggestion
                except Exception as e:
                    logger.error(f"Failed to download {content_file.download_url}: {e}")
                    raise e
```

Reply:
Done.

Reply:
Sorry this was my fault when I commented, this should be: 
```python
logger.error(f"Failed to download {path}: {e}")
```

Since the `content_file` might not exist if the error is raised.

Reply:
Also the above code was never updated?

Reply:
Done